### PR TITLE
feat: [CCM-5464]: node recommendation enhancements

### PR DESCRIPTION
**Pull request by rajarshee-harness**
##### Jira Links:

https://harness.atlassian.net/browse/CCM-5464

#### After:

Screenshot 2022-03-09 at 5 52 05 PM

#### Before:

Screenshot 2022-03-09 at 5 51 27 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by rajarshee-harness** [comment]
trigger rebase

**Comment by rajarshee-harness** [comment]
trigger rebase

File: src/modules/75-ce/components/NodeRecommendation/TuneNodeRecommendationCard.tsx [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+          min={0}
+          max={100}
+          stepSize={1}
+          labelRenderer={false}
+          value={buffer}
+          onChange={val => setBuffer(val)}
+          className={css.bufferSlider}
+        />
+        <Container>
+          <Text inline font={{ variation: FontVariation.SMALL }}>{`${getString(
```
Comment (jenil-harness) [inline src/modules/75-ce/components/NodeRecommendation/TuneNodeRecommendationCard.tsx:214]: knit: you can define one string and pass variables to it.

**Review by jenil-harness — CHANGES_REQUESTED** [review]

File: src/modules/75-ce/components/InstanceFamiliesModalTab/InstanceFamiliesModalTab.module.scss [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    z-index: 9999;
+  }
+
+  .hoverBackground {
+    background-color: rgba(2, 120, 213, 0.05) !important;
+  }
+
+  tbody {
+    tr:nth-child(odd) {
+      background-color: rgba(27, 46, 73, 0.02);
```
[the remark this answers is not in the document]
Comment (rajarshee-harness) [reply src/modules/75-ce/components/InstanceFamiliesModalTab/InstanceFamiliesModalTab.module.scss:32]: We have variables for both the colours but as they have opacity of 0.05 and 0.02 and adding opacity affects the entire component, I am using rgba.

File: src/modules/75-ce/utils/momentUtils.ts [hunk 3]
```diff
@@ -18,6 +18,14 @@ export const GET_DATE_RANGE = {
     todayInUTC().subtract(1, 'days').format()
   ]
 }
+export const GET_NODEPOOL_DATE_RANGE = {
+  [NodepoolTimeRangeType.LAST_DAY]: [todayInUTC().subtract(1, 'days').startOf('day').format(), todayInUTC().format()],
```
[the remark this answers is not in the document]
Comment (rajarshee-harness) [reply src/modules/75-ce/utils/momentUtils.ts:22]: There were constants for timerange but they didn’t have `LAST_DAY` and I wasn't sure how adding it to them would affect the existing components so defined a new one for nodepool.

**Comment by rajarshee-harness** [comment]
retrigger typecheck

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
trigger rebase

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
retrigger jest

File: src/modules/75-ce/pages/node-recommendation-details/NodeRecommendationDetailsPage.tsx [hunk 4]
```diff
@@ -54,143 +57,96 @@ const NodeRecommendationDetailsPage = () => {
...
+              {nodePoolData.resourceName}
+            </Text>
+          </Layout.Horizontal>
+        }
+        breadcrumbs={<NGBreadcrumbs className={css.breadCrumb} links={breadCrumbLinks} />}
+      />
+      <Page.Body>
+        <Card style={{ width: '100%' }}>
+          <Layout.Horizontal spacing="small">
+            <Text color={Color.GREY_800} font={{ weight: 'semi-bold' }}>
```
[the remark this answers is not in the document]
Comment (jenil-harness) [reply src/modules/75-ce/pages/node-recommendation-details/NodeRecommendationDetailsPage.tsx:100]: ?

[the remark this answers is not in the document]
Comment (rajarshee-harness) [reply src/modules/75-ce/pages/node-recommendation-details/NodeRecommendationDetailsPage.tsx:100]: Forgot to edit it while copying the existing component. Will fix this.

File: src/modules/75-ce/components/NodeRecommendation/NodeRecommendation.tsx [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+
+  const pathParams = {
+    provider: defaultTo(provider, ''),
+    region: defaultTo(region, ''),
+    service: defaultTo(service, '')
+  }
+  const { mutate: fetchNewRecommendation, loading } = useRecommendCluster(pathParams)
+  const { data: seriesData, loading: seriesDataLoading } = useGetSeries(pathParams)
+
+  const debouncedFetchNewRecomm = useCallback(pDebounce(fetchNewRecommendation, 500), [])
```
[the remark this answers is not in the document]
Comment (rajarshee-harness) [reply src/modules/75-ce/components/NodeRecommendation/NodeRecommendation.tsx:186]: It didn't have a debounce option.

**Comment by rajarshee-harness** [comment]
trigger rebase

**Comment by rajarshee-harness** [comment]
trigger cypress

**Review by vkbansal — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]
approving the configs and package.json changes

**Review by jenil-harness — APPROVED** [review]

**Pull request merged**